File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(S, k):
    heapq.heapify(S)
    answer = 0
    while S[0]<k:    
        if len(S) < 2:
            return -1
        heapq.heappush(S,heapq.heappop(S) + (heapq.heappop(S) * 2))
        answer += 1
    return answer

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;

        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        for (int scov : scoville) {
            minHeap.add(scov);
        }

        while (minHeap.peek() < K) {
            if (minHeap.size() == 1) {
                return -1;
            }

            int check = minHeap.poll() + minHeap.poll() * 2;

            answer++;
            minHeap.add(check);
        }
        return answer;
    }
}
